.circleci/config.yml

```diff
@@ -0,0 +1,139 @@
+
+      - save_cache:
+          key: v1-py2-{{ checksum "setup.py" }}
+          paths:
+            - /tmp/venv/country_template
+
+      - run:
+          name: Run tests
+          command: make test
+
+      - run:
+          name: Check version number has been properly updated
+          command: |
+            git fetch
+            ./.circleci/check-version-bump.sh
+
+  deploy_python2:
+    docker:
+      - image: python:2.7.14
+    environment:
+      PYPI_USERNAME: openfisca-bot
+      # PYPI_PASSWORD: this value is set in CircleCI's web interface; do not set it here, it is a secret!
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          key: v1-py2-{{ checksum "setup.py" }}
+
+      - run:
+          name: Upload a Python 2 package to Pypi and publish a tag if functional changes were made
+          command: |
+            source /tmp/venv/country_template/bin/activate
+            ./.circleci/deploy-python-2.sh
+
+  build_python3:
+    docker:
+      - image: python:3.6
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          key: v1-py3-{{ checksum "setup.py" }}
+
+      - run:
+          name: Create a virtualenv
+          command: |
+            mkdir -p /tmp/venv/country_template
+            python -m venv /tmp/venv/country_template
+            echo "source /tmp/venv/country_template/bin/activate" >> $BASH_ENV
+
+      - run:
+          name: Install dependencies
+          command: |
+            pip install --upgrade pip twine wheel
+            pip install --editable .[test] --upgrade
+          #  pip install --editable git+https://github.com/openfisca/country-template.git@BRANCH_NAME#egg=OpenFisca-Country-Template  # use a specific branch of OpenFisca-Country-Template
+
+      - save_cache:
+          key: v1-py3-{{ checksum "setup.py" }}
+          paths:
+            - /tmp/venv/country_template
+
+      - run:
+          name: Run tests
+          command: make test
+
+      - run:
+          name: Check version number has been properly updated
+          command: |
+            git fetch
+            ./.circleci/check-version-bump.sh
+
+  deploy_python3:
+    docker:
+      - image: python:3.6
+    environment:
+      PYPI_USERNAME: openfisca-bot
+      # PYPI_PASSWORD: this value is set in CircleCI's web interface; do not set it here, it is a secret!
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          key: v1-py3-{{ checksum "setup.py" }}
+
+      - run:
+          name: Upload a Python 3 package to Pypi
+          command: |
+            source /tmp/venv/country_template/bin/activate
+            ./.circleci/deploy-python-3.sh
+
+workflows:
+  version: 2
+  build_and_deploy:
+    jobs:
+      - build_python2
+      - build_python3
+      - deploy_python2:
+          requires:  # Only deploy if tests passed both with Python 2 and Python 3
```

Comment:
> This is quite readable in the config. This comment would add more value by explaining _why_ we don't want to deploy unless both versions pass.

Reply:
Ok, I thought it was unclear from a previous comment. Do you think we should not require the `build_python2` for  `deploy_python3`? I guess that we are on master, so technically the tests have already passed...

Reply:
Removed the dependency, as it's not really necessary

Reply:
I thought it would be because we don't really want to publish a new version for only one version: I wouldn't want to deal with version `23.1.2` existing in Python3 but not in Python2 and the other way around for `23.1.3` 😅